securedrop_export/print/service.py

```diff
@@ -0,0 +1,285 @@
+                    logger.info("Print completed")
+                    return True
+                else:
+                    time.sleep(5)
+            except subprocess.CalledProcessError:
+                raise ExportException(sdstatus=Status.ERROR_PRINT)
+            except TimeoutException:
+                logger.error("Timeout waiting for printer {}".format(self.printer_name))
+                raise ExportException(sdstatus=Status.ERROR_PRINT)
+        return True
+
+    def _check_printer_setup(self) -> None:
+        """
+        Check printer setup.
+        Raise ExportException if supported setup is not found.
+        """
+        try:
+            logger.info("Searching for printer")
+            output = subprocess.check_output(["sudo", "lpinfo", "-v"])
+            printers = [x for x in output.decode("utf-8").split() if "usb://" in x]
+            if not printers:
+                logger.info("No usb printers connected")
+                raise ExportException(sdstatus=Status.ERROR_PRINTER_NOT_FOUND)
+
+            supported_printers = [
+                p for p in printers if any(sub in p for sub in ("Brother", "LaserJet"))
+            ]
+            if not supported_printers:
+                logger.info("{} are unsupported printers".format(printers))
+                raise ExportException(sdstatus=Status.ERROR_PRINTER_NOT_SUPPORTED)
+
+            if len(supported_printers) > 1:
+                logger.info("Too many usb printers connected")
+                raise ExportException(sdstatus=Status.ERROR_MULTIPLE_PRINTERS_FOUND)
+
+            printer_uri = printers[0]
+            printer_ppd = self._install_printer_ppd(printer_uri)
+            self._setup_printer(printer_uri, printer_ppd)
+        except subprocess.CalledProcessError as e:
+            logger.error(e)
+            raise ExportException(sdstatus=Status.ERROR_UNKNOWN)
+
+    def _get_printer_uri(self) -> str:
+        """
+        Get the URI via lpinfo. Only accept URIs of supported printers.
+
+        Raise ExportException if supported setup is not found.
+        """
+        printer_uri = ""
+        try:
+            output = subprocess.check_output(["sudo", "lpinfo", "-v"])
+        except subprocess.CalledProcessError:
+            logger.error("Error attempting to retrieve printer uri with lpinfo")
+            raise ExportException(sdstatus=Status.ERROR_PRINTER_URI)
+
+        # fetch the usb printer uri
+        for line in output.split():
+            if "usb://" in line.decode("utf-8"):
+                printer_uri = line.decode("utf-8")
+                logger.info("lpinfo usb printer: {}".format(printer_uri))
+
+        # verify that the printer is supported, else throw
+        if printer_uri == "":
+            # No usb printer is connected
+            logger.info("No usb printers connected")
+            raise ExportException(sdstatus=Status.ERROR_PRINTER_NOT_FOUND)
+        elif not any(x in printer_uri for x in ("Brother", "LaserJet")):
```

Comment:
Nit: `("Brother", "LaserJet")` is used at least thrice, it could be extracted to a constant.

Reply:
thank you, addressed in https://github.com/freedomofpress/securedrop-export/pull/105/commits/282a9a085c5fecfb301663bce9d148ea7e7a9f60